src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/ReleaseDateService.kt

```diff
@@ -15,29 +17,40 @@ class ReleaseDateService(
   @Value("\${maxNumberOfWorkingDaysToUpdateLicenceTimeOutStatus:3}") private val maxNumberOfWorkingDaysToUpdateLicenceTimeOutStatus: Int,
 ) {
 
-  fun getCutOffDateForLicenceTimeOut(jobExecutionDate: LocalDate): LocalDate {
-    return generateSequence(jobExecutionDate) { it.plusDays(1) }
-      .filterNot { excludeBankHolidaysAndWeekends(it) }
+  fun getCutOffDateForLicenceTimeOut(now: Clock? = null): LocalDate {
+    return generateSequence(LocalDate.now(now ?: clock)) { it.plusDays(1) }
+      .filterNot { isBankHolidayOrWeekend(it) }
       .take(maxNumberOfWorkingDaysToUpdateLicenceTimeOutStatus)
       .last()
   }
 
+  fun isInHardStopPeriod(licence: Licence, now: Clock? = null): Boolean {
+    val timedOutDate = getCutOffDateForLicenceTimeOut(now)
+    val releaseDate = licence.actualReleaseDate ?: licence.conditionalReleaseDate
+
+    if (releaseDate == null) {
+      log.warn("Licence with id: ${licence.id} has no CRD or ARD")
+      return false
+    }
+    return releaseDate <= timedOutDate
+  }
```

Comment:
new function!

Reply:
Forgive me if I'm misunderstanding, is this not saying that it will return true if the licence release date is before or equal to the hard-stop window start? ie it would be true from licence creation up to 1 day before release?

If I'm understanding correctly, I believe it should be the inverse:
`return timedOutDate <= releaseDate && releaseDate >= now`

Reply:
Very minor, non-blocking, and if others disagree please ignore

Could we rename `timedOutDate` to `endOfHardStopPeriod` or similar?